djangosaml2/utils.py

```diff
@@ -99,6 +100,24 @@ def get_fallback_login_redirect_url():
 
 
 def validate_referral_url(request, url):
+    # Ensure the url is even a valid URL; sometimes the given url is a
+    # RelayState containing PySAML data.
+    # Some technically-valid urls will be fail this check, so the
+    # SAML_STRICT_URL_VALIDATION setting can be used to turn off this check.
+    # This should only happen if there is no slash, host and/or protocol in the
+    # given URL. A better fix would be to add those to the RelayState.
+    saml_strict_url_validation = getattr(
+        settings,
+        "SAML_STRICT_URL_VALIDATION",
+        True
+    )
+    try:
+        if saml_strict_url_validation:
+            url = resolve_url(url)
```

Comment:
This will also enable Django URLconf names to be passed as `RelayState` values as well. It's not documented, but I don't think this will cause problems. 

If this is undesirable, it would be easy to disable this behavior

Reply:
I'd suggest to get the redention by documenting it

Reply:
I hope the changes in [64f6bce](https://github.com/IdentityPython/djangosaml2/pull/388/commits/64f6bcee47bb6dfecf9d50fdc17cbd1652d97fe2) is sufficient?